File: dgs-webflux/src/main/java/com/example/demo/Post.java
package com.example.demo;

import lombok.Builder;
import lombok.Data;
import lombok.ToString;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@ToString
public class Post {
    String id;
    String title;
    String content;

    @Builder.Default
    List<Comment> comments = new ArrayList<>();
    String authorId;
    Author author;
}

File: graphql-kotlin/src/main/kotlin/com/example/demo/gql/types/Post.kt
package com.example.demo.gql.types

import com.example.demo.gql.datafetchers.AuthorsDataLoader
import com.example.demo.gql.datafetchers.CommentsDataLoader
import com.example.demo.gql.directives.UpperCase
import com.expediagroup.graphql.server.extensions.getValueFromDataLoader
import graphql.schema.DataFetchingEnvironment
import java.time.LocalDateTime
import java.util.*
import java.util.concurrent.CompletableFuture

data class Post(
    val id: UUID?,
    @UpperCase
    val title: String,
    val content: String?,
    val status: String? = null,
    val createdAt: LocalDateTime?,
    val authorId: UUID? = null,
) {
    fun comments(environment: DataFetchingEnvironment): CompletableFuture<List<Comment>> {
        return environment.getValueFromDataLoader(CommentsDataLoader.NAME, id)
    }

    fun author(environment: DataFetchingEnvironment): CompletableFuture<Author> {
        return environment.getValueFromDataLoader(AuthorsDataLoader.NAME, authorId)
    }
}